dvc/repo/brancher.py

```diff
@@ -72,7 +78,9 @@ def brancher(  # noqa: E302
             self.root_dir = self.fs.path.join("/", *repo_root_parts)
 
             if cwd_parts:
-                cwd = self.fs.path.join("/", *cwd_parts)
+                cwd = self.fs.path.join(  # type: ignore[unreachable]
```

Comment:
It looks like it is an issue of `mypy`. I tried some tests and found that these parts of the code can be reached.